File: project/app/src/main/java/org/owntracks/android/injection/modules/ObjectboxWaypointsModule.java
package org.owntracks.android.injection.modules;

import android.content.Context;

import org.greenrobot.eventbus.EventBus;
import org.owntracks.android.data.repos.ObjectboxWaypointsRepo;
import org.owntracks.android.data.repos.WaypointsRepo;
import org.owntracks.android.injection.qualifier.AppContext;
import org.owntracks.android.support.Preferences;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ObjectboxWaypointsModule {
    @Provides
    @Singleton
    protected WaypointsRepo provideWaypointsRepo(@AppContext Context context, EventBus eventBus, Preferences preferences) {
        return new ObjectboxWaypointsRepo(context, eventBus, preferences);
    }
}

File: project/app/src/main/java/org/owntracks/android/data/repos/ObjectboxWaypointsModule.kt
package org.owntracks.android.data.repos

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

@InstallIn(SingletonComponent::class)
@Module
abstract class ObjectboxWaypointsModule {
    @Binds
    abstract fun bindWaypointsRepo(objectboxWaypointsRepo: ObjectboxWaypointsRepo): WaypointsRepo
}

